tests/test_x509.py

```diff
@@ -856,6 +856,38 @@ def test_add_unsupported_extension(self, backend):
                 critical=False,
             )
 
+    def test_add_unsupported_extension_via_init(self, backend):
+        private_key = RSA_KEY_2048.private_key(backend)
+
+        builder = x509.CertificateSigningRequestBuilder(
+            subject_name=x509.Name([
+                x509.NameAttribute(x509.OID_COUNTRY_NAME, u'US'),
+            ]),
+            extensions=[
+                x509.Extension(
+                    x509.OID_AUTHORITY_KEY_IDENTIFIER,
+                    False,
+                    x509.AuthorityKeyIdentifier('keyid', None, None)
+                )
+            ]
+        )
```

Comment:
So, a straightforward way would to be to add support for an extension to
the frontend, but not to the backend.

On Fri, Jun 26, 2015 at 7:30 PM, Paul Kehrer <notifications@github.com>
wrote:

> In tests/test_x509.py
> <https://github.com/pyca/cryptography/pull/2066#discussion_r33404877>:
>
> > @@ -856,6 +856,38 @@ def test_add_unsupported_extension(self, backend):
> >                  critical=False,
> >              )
> >
> > +    def test_add_unsupported_extension_via_init(self, backend):
> > +        private_key = RSA_KEY_2048.private_key(backend)
> > +
> > +        builder = x509.CertificateSigningRequestBuilder(
> > +            subject_name=x509.Name([
> > +                x509.NameAttribute(x509.OID_COUNTRY_NAME, u'US'),
> > +            ]),
> > +            extensions=[
> > +                x509.Extension(
> > +                    x509.OID_AUTHORITY_KEY_IDENTIFIER,
> > +                    False,
> > +                    x509.AuthorityKeyIdentifier('keyid', None, None)
> > +                )
> > +            ]
> > +        )
>
> If you have a better way to trigger the underlying error in the backend
> I'm happy to change it. If I don't do this I don't have a way to trigger
> the NotImplementedError in the backend since it will catch it inside the
> builder before it ever gets that far.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2066/files#r33404877>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084